lib/base/logger.cpp

```diff
@@ -243,17 +244,35 @@ void Logger::UpdateMinLogSeverity()
 	m_MinLogSeverity.store(result);
 }
 
-Log::Log(LogSeverity severity, String facility, const String& message)
-	: Log(severity, std::move(facility))
+Log::Log(LogSeverity severity, String facility, const String& message, const std::vector<ConfigObject::Ptr>& involved)
+	: Log(severity, std::move(facility), involved)
 {
 	if (!m_IsNoOp) {
 		m_Buffer << message;
 	}
 }
 
-Log::Log(LogSeverity severity, String facility)
-	: m_Severity(severity), m_Facility(std::move(facility)), m_IsNoOp(severity < Logger::GetMinLogSeverity())
-{ }
+Log::Log(LogSeverity severity, String facility, const std::vector<ConfigObject::Ptr>& involved)
+	: m_Severity(severity), m_Facility(std::move(facility))
+{
+	std::set<ConfigObject::Ptr> objects;
+
+	for (auto& obj : involved) {
+		objects.merge(DependencyGraph::GetAllParents(obj));
+	}
+
+	objects.insert(involved.begin(), involved.end());
+
+	for (auto& obj : objects) {
+		auto severity (obj->GetOverrideSeverityRaw());
+
+		if (severity >= 0 && severity < m_OverriddenMinSeverity) {
+			m_OverriddenMinSeverity = (LogSeverity)severity;
+		}
+	}
+
+	m_IsNoOp = severity < std::min(Logger::GetMinLogSeverity(), m_OverriddenMinSeverity);
```

Comment:
This loop would still exist.

Reply:
Then I don't quite understand why the parameter is `const std::vector<ConfigObject::Ptr>& involved` is at the moment (i.e. a vector instead of a single pointer). Looks like you want to specify multiple objects (where with the current code, the parents for each one would be looked up). Is there a situation where this is necessary (or at least useful)?

Reply:
I didn't examine a single caller yet, not to mention all of them. If it turns out that no log message immediately involves multiple not necessarily related objects, I'll omit the vector.